app/models/event_stream.rb

```diff
@@ -22,4 +22,6 @@ class EventStream < ApplicationRecord
   belongs_to :container_replicator
   belongs_to :container_group
   belongs_to :container_node
+
+  belongs_to :middleware_server, :foreign_key => :middleware_server_id
```

Comment:
Probably it is not needed.
Once #10335 is merged I will set up an end-to-end integration scenario and update it if necessary.
An integration test for this feature requires several levels of mockups (not only VCR) and those are not including yet.
They will be provided in a following PR.